interface/ceed-operator.c

```diff
@@ -88,28 +88,57 @@ static int CeedOperatorCheckField(Ceed ceed, CeedQFunctionField qf_field,
 
   }
   // Field size
+  CeedFESpace FE_space = b->basis_space;
   switch(eval_mode) {
   case CEED_EVAL_NONE:
-    if (size != restr_num_comp)
-      // LCOV_EXCL_START
-      return CeedError(ceed, CEED_ERROR_DIMENSION,
-                       "Field '%s' of size %" CeedInt_FMT " and EvalMode %s: ElemRestriction has "
-                       CeedInt_FMT " components",
-                       qf_field->field_name, qf_field->size, CeedEvalModes[qf_field->eval_mode],
-                       restr_num_comp);
-    // LCOV_EXCL_STOP
+    switch (FE_space) {
```

Comment:
CEED_EVAL_NONE will only occur with CEED_BASIS_COLLOCATED, so there is no need for special case here

Reply:
Correct, thanks. So I guess instead of the switch case I can multiply the num_comp by Q_comp in interpolation mode, right?

Reply:
Yeah, I tried and worked. I use Q_comp, it is cleaner